src/Compilers/VisualBasic/Portable/Symbols/Tuples/TupleErrorFieldSymbol.vb

```diff
@@ -94,7 +94,10 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
 
         Public Overloads Function Equals(other As TupleErrorFieldSymbol) As Boolean
             Return other Is Me OrElse
-                (other IsNot Nothing AndAlso Me._tupleFieldId = other._tupleFieldId AndAlso Me.ContainingType = other.ContainingType)
+                (other IsNot Nothing AndAlso
+                Me._tupleElementIndex = other._tupleElementIndex AndAlso
+                Me.ContainingType = other.ContainingType AndAlso
+                Me.Name = other.Name)
```

Comment:
An idea. Does not need to go in this change. -
Make internal encoding of element indices like:
2 * N  for ItemN default elements, that are there regardless whether defined in source or not.  
2 * N + 1 for user defined aliases like "Alice", "Bob"

That will make the internal index value unique within the containing tuple and we would only need to compare names when creating fields. No need to do name compare in equality.

This encoding would also make it possible to get an actual index of a field in a tuple.
In the possible public API, we could return ```(int?)null```   for nonelement fields with negative _tupleFieldId , and return ```_tupleFieldId >> 1``` for elements.